### DISCUSSION TITLE
Bitnami Airflow chart 19.0.3 (Airflow version 2.10) not creating pods with unique names

### DISCUSSION THREAD

**Discussion by Arunkumar-sai** [apache/airflow#41885] [Q&A]
### Apache Airflow version

2.10.0

### What happened?

Pods are not getting unique names. In the earlier version of the chart we used to get unique pod names. but after upgrading to the latest chart, its not generating random pod names, instead its trying to use the metadata.name value as the pod name and i am facing issue when try to create multiple pods

### What you think should happen instead?

Pods should get unique names

### How to reproduce

Upgrade to Bitnami chart version from 13.1.1 to 19.0.3

### Operating System

Debian GNU/Linux 12

### Versions of Apache Airflow Providers

2.10

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

Helm deployment

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by Arunkumar-sai** [comment]
still didn't figured out the issue or root cause

**Comment by jscheffl** [comment]
If you are using the Bitnami Helm Chart - are you sure this is the right issue tracker for the problem? Bitnami Helm CHart is not maintained by the Airflow repo.

**Comment by Arunkumar-sai** [comment]
@jscheffl - can you help to understand why airflow is not updating the pod metadata.name, its creating with the exact same name we use in the pod_template.yaml file. are we suppose to do anything in addition to get the unique pod names

**Comment by jscheffl** [comment]
I believe there is a lot of context missing. Which executor are you using? Is it about the POD names of Airflow or the workload? How is your DAG code looking like? How is your Kubernetes config looking like?

And most prominent: If it was working before - which other parameters have you adjusted that so that it was working before?

If you found a bug, then it is okay to post it. But this is not a research forum.

**Comment by Arunkumar-sai** [comment]
My apologies, i would have raised under Q&A.

I am using the kubernetes executor. and its about POD names for the workloads

DAG:
from datetime import datetime, timedelta
from airflow.models.dag import DAG
from airflow.operators.dummy_operator import DummyOperator
from airflow.utils.dates import days_ago
from airflow.operators.bash_operator import BashOperator
import json
import os
import sys

sys.path.append(os.path.dirname(os.path.realpath(__file__)))

#Utils
import dag_utils

now = datetime.now()
now_to_the_hour = (
    now - timedelta(0, 0, 0, 0, 0, 3)
).replace(minute=0, second=0, microsecond=0)
START_DATE = now_to_the_hour
DAG_NAME = 'TEST_COMMANDS'

default_args = {
    'owner': 'airflow',
    'depends_on_past': False,
    'start_date': days_ago(2)
}

executor_config = test_akr_dag_utils.executor('test')

dag = DAG(DAG_NAME, default_args=default_args,schedule_interval='@once')
test_cmd = BashOperator(
    task_id="test_cmd",
    executor_config=executor_config,
    bash_command=". $HOME/.bashrc ; set; echo 'the panel value: ';  echo $TEST_IMAGE_TAG; echo $SHAW_OFFSET;  echo $STB_NGV_TUNES_OUTPUT; sleep 259200",
    dag=dag)

Executor yaml

    container:
      test:
        apiVersion: v1
        kind: Pod
        metadata:
          name: test 
        spec:
          containers:
          - name: base
            image: test.image
            imagePullPolicy: {{ .image.pullPolicy }}
            securityContext:
             runAsUser: 1000
            volumeMounts:  
            - name: test-data 
              mountPath: /home/akr/batch/data 
              readOnly: False
            - name: test-sftp
              mountPath: /home/akr/batch/test-sftp
              readOnly: {{ .Values.sftp.vmounts.readonly }}
            envFrom:
            - configMapRef:
                name: test-config
            - configMapRef:
                name: test-akr-airflow-env
            - secretRef:
                name: test-db
            - secretRef:
                name: test-mail
          volumes:
          - name: test-data 
            persistentVolumeClaim:
              claimName: pvc-test-data
